cmake/backports/FindPythonInterp.cmake

```
@@ -171,6 +171,8 @@ if(PYTHON_EXECUTABLE)
 endif()
 
 include(FindPackageHandleStandardArgs)
+set(FPHSA_NAME_MISMATCHED 1)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(PythonInterp REQUIRED_VARS PYTHON_EXECUTABLE VERSION_VAR PYTHON_VERSION_STRING)
+unset(FPHSA_NAME_MISMATCHED)
```

Although I don't expect this to be an issue in this case, then in general it is safer to read existing value, and restore that afterwards.
```suggestion
set(FPHSA_NAME_MISMATCHED_CURRENT ${FPHSA_NAME_MISMATCHED})
set(FPHSA_NAME_MISMATCHED 1)
FIND_PACKAGE_HANDLE_STANDARD_ARGS(PythonInterp REQUIRED_VARS PYTHON_EXECUTABLE VERSION_VAR PYTHON_VERSION_STRING)
set(FPHSA_NAME_MISMATCHED ${FPHSA_NAME_MISMATCHED_CURRENT})
```

---

Yes, that is better. Done.